### PR TITLE
First commit

### PR DESCRIPTION
**Pull request by SanduBogdan95**

### CODE REVIEW

File: src/test/java/teofilursan/tests/seleniumpart1/SeleniumExampleTest.java [hunk 1]
```diff
@@ -1,22 +0,0 @@
-package teofilursan.tests.seleniumpart1;
```
Comment (95teo) [inline src/test/java/teofilursan/tests/seleniumpart1/SeleniumExampleTest.java:1]: Please don't remove this class yet. Thanks!

File: src/test/java/BogdanSandu/tests/Tema6/HerokuForms.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    public static void navigateToHerokuFormsPage() {
+        driver = BrowserManager.createDriverAndGetPage();
+        driver.get("https://testpages.herokuapp.com/styled/basic-html-form-test.html");
+        System.out.println("Am deschis Heroku Forms page!");
+    }
+
+    public static void scrollToSubmitButton() {
+        WebElement submitButton = driver.findElement(By
```
Comment (95teo) [inline src/test/java/BogdanSandu/tests/Tema6/HerokuForms.java:26]: Please complete some form details before clicking on submit button. Thanks!

File: src/test/java/BogdanSandu/tests/Tema6/HerokuForms.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import BogdanSandu.driver.BrowserManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.interactions.Actions;
+
+public class HerokuForms {
+
+    static ChromeDriver driver = null;
+
```
Comment (95teo) [inline src/test/java/BogdanSandu/tests/Tema6/HerokuForms.java:12]: You are missing the try - catch block. Please add it.

File: src/test/java/BogdanSandu/tests/Tema6/DemoQaWindow.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+import BogdanSandu.driver.BrowserManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+import java.util.Set;
+
+public class DemoQaWindow {
+
```
Comment (95teo) [inline src/test/java/BogdanSandu/tests/Tema6/DemoQaWindow.java:11]: You are missing the try - catch block. Please add it.

File: src/test/java/BogdanSandu/tests/Tema6/DemoQaWindow.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        clickOnNewTabButton();
+        closeBrowser();
+    }
+
+    public static void navigateToDemoQAWindowPage() {
+        driver = BrowserManager.createDriverAndGetPage();
+        driver.get("https://demoqa.com/browser-windows");
+        System.out.println("Am deschis Demo QA window page!");
+    }
+
```
Comment (95teo) [inline src/test/java/BogdanSandu/tests/Tema6/DemoQaWindow.java:25]: Add interactions with other buttons too - there are two more on this page, not only the one showed at retro. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

File: src/test/java/BogdanSandu/tests/Tema6/DemoQaWindow.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import java.util.Set;
+
+public class DemoQaWindow {
+
+    static ChromeDriver driver = null;
+
+    public static void main(String[] args) {
+        navigateToDemoQAWindowPage();
+        try {
+            clickOnNewTabButton();
```
Comment (95teo) [inline src/test/java/BogdanSandu/tests/Tema6/DemoQaWindow.java:19]: Do you have any exceptions that can be thrown inside this try? That means to identify in one of your methods an element with a wrong id/selector/xpath. ;)

File: src/test/java/BogdanSandu/tests/Tema6/HerokuForms.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+import static BogdanSandu.tests.Tema6.HerokuAlerts.tryToFindElement;
+
+public class HerokuForms {
+
+    static ChromeDriver driver = null;
+
+    public static void main(String[] args) {
+        navigateToHerokuFormsPage();
+        try {
+            fillFormDetails();
```
Comment (95teo) [inline src/test/java/BogdanSandu/tests/Tema6/HerokuForms.java:27]: Do you have any exceptions that can be thrown inside this try? That means to identify in one of your methods an element with a wrong id/selector/xpath. ;)

File: src/test/java/BogdanSandu/tests/Tema6/DemoQaWindow.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+                System.out.println("Text de pe noul window " + newWindowHeading.getText());
+                driver.close();
+            }
+        }
+        driver.switchTo().window(parentTab);
+        System.out.println("Am facut click pe new window button");
+    }
+
+
+    public static void closeBrowser() {
```
Comment (95teo) [inline src/test/java/BogdanSandu/tests/Tema6/DemoQaWindow.java:78]: No need to declare here one extra closeBrowser() method. You can use the one from BrowserManager. Thanks!

File: src/test/java/BogdanSandu/tests/Tema6/HerokuAlerts.java [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+    public static void tryToFindElement(){
+        try {
+            WebElement tryFind = driver.findElement(By.id("Boo-hoo"));
+        }catch (NoSuchElementException e){
+            FileUtils.takeScreenshot(driver, "inexistentElement");
+            System.out.println("Nu exista elementul cautat!");
+        }
+    }
+
+    public static void closeBrwoser(){
```
Comment (95teo) [inline src/test/java/BogdanSandu/tests/Tema6/HerokuAlerts.java:147]: No need to declare here one extra closeBrowser() method. You can use the one from BrowserManager. Thanks!

File: src/test/java/BogdanSandu/tests/Tema6/HerokuForms.java [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+    public static void scrollToSubmitButton () {
+        WebElement submitButton = driver.findElement(By.xpath("//input[@type='submit']"));
+        Actions actions = new Actions(driver);
+        actions.moveToElement(submitButton).build().perform();
+        submitButton.click();
+        System.out.println("Am facut scroll catre submit button");
+    }
+
+    public static void closeBrowser () {
```
Comment (95teo) [inline src/test/java/BogdanSandu/tests/Tema6/HerokuForms.java:79]: No need to declare here one extra closeBrowser() method. You can use the one from BrowserManager. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

**Review by 95teo — CHANGES_REQUESTED** [review]

File: src/main/java/bogdanSandu/driver/BrowserManager.java [hunk 11]
```diff
@@ -1,4 +1,4 @@
-package bogdanSandu.driver;
+package BogdanSandu.driver;
```
Comment (95teo) [inline src/main/java/bogdanSandu/driver/BrowserManager.java:1]: Packages in Java should start with lower and not capital letters so please leave the package name to bogdanSandu. Also do this change for the package inside test/java. Thanks!

File: src/test/java/BogdanSandu/tests/Tema6/DemoQaWindow.java [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+public class DemoQaWindow {
+
+    static ChromeDriver driver = null;
+
+    public static void main(String[] args) {
+        navigateToDemoQAWindowPage();
+        try {
+            clickOnNewTabButton();
+            clickOnNewWindow();
+            FileUtils.takeScreenshot(driver, "alertDemoQa");
```
Comment (95teo) [inline src/test/java/BogdanSandu/tests/Tema6/DemoQaWindow.java:22]: Remove this line too as it's not needed. Screenshot should be taken only when getting an Exception that will cause test case to fail. Thanks!

File: src/test/java/BogdanSandu/tests/Tema7/TestDrive.java [hunk 13]
```diff
@@ -0,0 +1,4 @@
+package bogdanSandu.tests.Tema7;
```
Comment (95teo) [inline src/test/java/BogdanSandu/tests/Tema7/TestDrive.java:1]: Please remove this class from this PR. You'll add it in PR for homework 7. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

File: src/main/java/bogdanSandu/driver/BrowserManager.java (hunk 11 above)
Comment (95teo) [reply src/main/java/bogdanSandu/driver/BrowserManager.java:1]: Please use lowerCase when naming packages as per Java conventions. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]
Please use lowerCase when naming packages as per Java conventions. Thanks!

File: src/main/java/bogdanSandu/driver/BrowserManager.java (hunk 11 above)
Comment (95teo) [inline src/main/java/bogdanSandu/driver/BrowserManager.java:1]: Here it seems to be package with upperCase letter. Can you please check again?

File: src/main/java/bogdanSandu/utils/FileUtils.java [hunk 14]
```diff
@@ -0,0 +1,20 @@
+package BogdanSandu.utils;
```
Comment (95teo) [inline src/main/java/bogdanSandu/utils/FileUtils.java:1]: Here it seems to be package with upperCase letter. Can you please check again?

**Review by 95teo — CHANGES_REQUESTED** [review]

**Review by redandreea — APPROVED** [review]

**Pull request merged**